node/hack/repair_terra/repair.go

```diff
@@ -47,41 +47,42 @@ func getAdminClient(ctx context.Context, addr string) (*grpc.ClientConn, error,
 	return conn, err, c
 }
 
-func getSequenceForTxhash(txhash string) (uint64, error) {
-	client := &http.Client{ 
-		Timeout: time.Second * 5, 
+func getSequencesForTxhash(txhash string) ([]uint64, error) {
+	client := &http.Client{
+		Timeout: time.Second * 5,
 	}
 	url := fmt.Sprintf("https://fcd.terra.dev/cosmos/tx/v1beta1/txs/%s", txhash)
 	resp, err := client.Get(url)
 	if err != nil {
-		return 0, fmt.Errorf("failed to get message: %w", err)
+		return []uint64{}, fmt.Errorf("failed to get message: %w", err)
 	}
 	defer resp.Body.Close()
 	txBody, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return 0, fmt.Errorf("failed to read message: %w", err)
+		return []uint64{}, fmt.Errorf("failed to read message: %w", err)
 	}
 
 	txJSON := string(txBody)
 	if !gjson.Valid(txJSON) {
-		return 0, fmt.Errorf("invalid JSON response")
+		return []uint64{}, fmt.Errorf("invalid JSON response")
 	}
 	txHashRaw := gjson.Get(txJSON, "tx_response.txhash")
 	if !txHashRaw.Exists() {
-		return 0, fmt.Errorf("terra tx does not have tx hash")
+		return []uint64{}, fmt.Errorf("terra tx does not have tx hash")
 	}
 	txHash := txHashRaw.String()
 
 	events := gjson.Get(txJSON, "tx_response.events")
 	if !events.Exists() {
-		return 0, fmt.Errorf("terra tx has no events")
+		return []uint64{}, fmt.Errorf("terra tx has no events")
 	}
 	msgs := EventsToMessagePublications(*terraAddr, txHash, events.Array())
 	// Should only ever be 1 message. Stole the above function from watcher.go
-	if len(msgs) != 1 {
-		return 0, fmt.Errorf("EventsToMessagePublications returned %d msgs", len(msgs))
+	var sequences = []uint64{}
+	for _, msg := range msgs {
+		sequences = append(sequences, msg.Sequence)
```

Comment:
We should filter out emitters that don't match so we don't exit earlier on lower sequence numbers (or incorrectly match up the wrong one). @SEJeff to take a look 🙂

Reply:
another thing we should do is make sure we only do a single reobservation request per transaction (even if it contains multiple sequence numbers)

Reply:
I implemented your suggestion @evan-gray in c3ae93dab87015d810baea41bcdad89c3c5066bf.

We can do your suggestion in a followup @kcsongor, but what we have works perfectly.

Reply:
I agree we don’t need to change further right now 🙂